src/App.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { Provider } from 'mobx-react';
+import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
+
+import './assets/styles/app.css';
+import { timelineStore, moduleInfoStore } from './store/index'; // STORE YOUR STORES AROUND HERE
+import Header from './components/Header/Header';
```

Comment:
So: your `App` component is the top-level component, which renders some "permanent" UI (header, footer), and then "dynamic" routes inside it.

Two thoughts:

1. Based on where your provider is, what if the `Header` or `Footer` need access to a store? I see a "user icon" image in the `Header`, for example.
2. It would be much more clear to split your components into two directories, eg `pages` and `components`. Right now you have everything in one directory, which is pretty confusing. The `App` component basically renders pages wrapped with a header and footer, so these page-level components might as well be called pages.

Reply:
Yes I was thinking about expanding the provider to include the header and the footer (don't know whether it's needed in the footer but there's no harm I think).
About 2 @isaachinman do you mean that all the being rendered components go in the `pages` and the rest of the components which are embedded within the ones that are being rendered go into `components`?

Reply:
Yes to your second question. In my opinion "pages" would be things that are the direct children of `App`, as they are the page content of each route.

Reply:
👍